### ISSUE TITLE
Long response time , gpu support?

### ISSUE THREAD

**Issue by drmetro09** [amithkoujalgi/ollama-pdf-bot#1]
Hi , im new to this i have installed this pdf chat bot by docker compose on unraid , but its not using gpu and taking 15 mins to 20mis to generate response any help and guidance on this ? I want to ask if this supports gpu nividia

**Comment by amithkoujalgi** [comment]
Hello @drmetro09. I have added a `docker-compose` setup for enabling GPU support for Ollama. Give that a try. 

If you're having an Nvidia GPU and you've installed the correct drivers, you should be able to use the GPU with the `docker-compose` setup from this repo.

PS: If you're setting up and running the setup for the very first time, you will need to wait for the model to download and the embedding model to load. Once these steps are done, the responses should be pretty fast.

Also, I've added benchmarks in the readme file for reference.

Happy exploring!

**Comment by drmetro09** [comment]
Hi , thanks for quick response but when gpu support is enabled it gives cublas error 15 and the bot doesn’t not give any response i have sent you a request on insta could you please guide me with this am new to this

**Comment by drmetro09** [comment]
![IMG_0678](https://github.com/amithkoujalgi/ollama-pdf-bot/assets/143595315/86dc247c-fcd5-44c3-bbc1-77c49791586d)

**Comment by amithkoujalgi** [comment]
The error is originating from the Ollama server module. Do you have the correct Nvidia drivers installed? If you have, could you share some screenshots for the same? And maybe share the details of your execution environment to get a better understanding about the issue.

You could have a look at the Ollama docker setup [documentation](https://hub.docker.com/r/ollama/ollama) to see if it helps you.

There seems to be a similar [issue](https://github.com/ggerganov/llama.cpp/issues/4477) reported in Ollama repo. See if you can correlate your issue to that.

Also,
> i have sent you a request on insta

I didn't quite understand this. :)

**Comment by drmetro09** [comment]
Yes , when I check the logs of Ollama container the logs show “cuBLAS error and than the pdf bot doesn’t give response . It’s working on cpu but on cpu it takes 20 to 30 mins for one response 
I use docker compose on UNRAID OS to deploy the containers both pdf bot and ollama both are on same network , i am using nvidia driver version : 535.146.02 , others like jellyfin plex etc work using gpu except the ollama .

have attached screenshots below
![IMG_0682](https://github.com/amithkoujalgi/ollama-pdf-bot/assets/143595315/a9303d56-37dc-4c7d-a7fb-cb931fd1a07d)
![IMG_0681](https://github.com/amithkoujalgi/ollama-pdf-bot/assets/143595315/70f9a2db-d209-482e-9a5b-217f84bda9a8)

**Comment by drmetro09** [comment]
This is docker compose yml , i want to know is it possible to limit the number of layers offloading to gpu by any env variable or anything , sorry im not a coder just trying to get this working 😄

![IMG_0683](https://github.com/amithkoujalgi/ollama-pdf-bot/assets/143595315/851c4f2f-5f18-4d42-b5d3-1e68ba8cb94e)

**Comment by drmetro09** [comment]
Have tried various ollama models like orca-mini , llama2, starling-lm all are behaving same

**Comment by drmetro09** [comment]
Update: Its finally working now , i downgraded from to v 0.1.10 and everything working fine its using the gpu without errors, tried using LLAMA2 model haven’t tried other models.
Thanks for your amazing work , i have few suggestions listed below , hope you would consider them for this project.
1) Increase size of pdf to be uploaded to more than 200mb maybe 600mb .
2) Option to include docx ,ppt , CSV , jpeg , jpg and other formats which are commonly used.
3) Option to give the bot an icon/Avatar of our choice .
4) Instead of rebuilding the vector database again when the same pdf is uploaded , the bot should have ability to store the previous database and use it again subsequently . This would save time and makes the output fast.
5) Ability to download and select various ollama models from the web UI of pdf bot 
6) using the bot for general chat besides the docs QnA.
( There is an amazing repo Private GPT for inspiration which satisfies the above points but its very complex to install and run from a perspective of non-IT guy )

**Comment by amithkoujalgi** [comment]
I guess there's not enough information for me to help you out with this. 
Consider opening an issue [here](https://github.com/ggerganov/llama.cpp/issues).
The Ollama authors might certainly be able to help you out.
Cheers!
